src/helpers.cc

```diff
@@ -467,6 +467,29 @@ std::pair<bool, Uptane::Target> LiteClient::downloadImage(const Uptane::Target&
   return {success, target};
 }
 
+void LiteClient::reportAktualizrConfiguration() {
+  if (!config.telemetry.report_config) {
+    LOG_DEBUG << "Not reporting libaktualizr configuration because telemetry is disabled";
+    return;
+  }
+
+  std::stringstream conf_ss;
+  config.writeToStream(conf_ss);
+  const std::string conf_str = conf_ss.str();
+  const Hash new_hash = Hash::generate(Hash::Type::kSha256, conf_str);
+  std::string stored_hash;
+  if (!(storage->loadDeviceDataHash("configuration", &stored_hash) &&
+        new_hash == Hash(Hash::Type::kSha256, stored_hash))) {
```

Comment:
`if (storage->loadDeviceDataHash("configuration", &stored_hash) && new_hash == Hash(Hash::Type::kSha256, stored_hash)) { LOG <<; return;}` 
IMO, the code will be slightly cleaner if we return if the config has not changed.

Reply:
Sure. I just took the code as-is from sotauptaneclient.cc.

Reply:
So i've updated the current first commit of this PR to just do like our other report* methods do. I can then do follow up commits to clean this code.

Reply:
I've tried to clean things up with the latest push. Let me know what you think